### PR TITLE
[FIX+IMP] delivery_multi_destination: Don't show children + demo

### PR DESCRIPTION
**Pull request by pedrobaeza**
* Don't show children carriers on many2one selections
* Don't search by default children carriers
* Add demo data

@Tecnativa

### CODE REVIEW

**Review by chienandalu — COMMENTED** [review]
Shouldn't by default these be hidden?
image

**Comment by pedrobaeza** [comment]
Yes, they should, but in local they were...

File: delivery_multi_destination/views/delivery_carrier_view.xml [hunk 1]
```diff
@@ -41,23 +41,23 @@
...
-                       invisible="not context.get('show_children_carriers', False)"
+                       invisible="not context.get('show_children_fields', False)"
                 />
             </field>
         </field>
     </record>
 
     <record id="delivery.action_delivery_carrier_form" model="ir.actions.act_window">
-        <field name="domain">[('parent_id', '=', False)]</field>
+        <field name="context">{'show_children_carriers': True}</field>
```
Comment (chienandalu) [inline delivery_multi_destination/views/delivery_carrier_view.xml]: When the menu option is hit this context is passed, and the default view is tree, so the children are visible in the tree. So I tried to set it to `False` here and to `True` in the form field that loads the tree.

File: delivery_multi_destination/views/delivery_carrier_view.xml [hunk 2]
```diff
@@ -27,7 +27,7 @@
                       attrs="{'invisible': [('destination_type', '!=', 'multi')]}"
                 >
                     <field name="child_ids"
-                           context="{'show_children_carriers': True}"
+                           context="{'show_children_fields': True}"
```
Comment (chienandalu) [inline delivery_multi_destination/views/delivery_carrier_view.xml:30]: Here a try to set `show_children_carriers` to `True` but seems that the action context prevails.

**Review by chienandalu — COMMENTED** [review]
I tried it locally and it behaves the same way. I tried to get it right with no luck. Seems to be some context priority issues.

File: delivery_multi_destination/views/delivery_carrier_view.xml (hunk 1 above)
Comment (pedrobaeza) [reply delivery_multi_destination/views/delivery_carrier_view.xml]: Solution is to keep both, domain and context. I have done it.

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Review by chienandalu — APPROVED** [review]
Perfect!

**Pull request merged**